app/controllers/articles_controller.rb

```diff
@@ -2,38 +2,55 @@
 
 class ArticlesController < ApplicationController
   before_action :set_article, only: %i[show edit update destroy]
+  before_action :set_user, only: %i[show edit update destroy]
 
   def index
-    @articles = Article.all
+    @articles = policy_scope(Article)
   end
 
   def new
     @article = Article.new
   end
 
   def edit
-    @user = current_user
+    authorize @article
+  rescue Pundit::NotAuthorizedError
+    flash[:alert] = "Cannot access the article #{@article.title}"
+    redirect_to user_articles_path
   end
 
-  def show; end
+  def show
+    @article = policy_scope(Article).find(params[:id])
```

Comment:
You can move ```policy_scope(Article)``` to ```set_article``` method, thanks that you will have
 ```suggestion
 def articles
    policy_scope(Article)
 end 
 def set_article
    @article = artilcles.find(params[:id])
 end 

```